custom_components/iec/coordinator.py

```diff
@@ -367,14 +367,9 @@ async def _insert_statistics(self, contract_id: int, is_smart_meter: bool) -> No
                 from_date = datetime.fromtimestamp(last_stat_time)
                 _LOGGER.debug(f"Last statistics are from {from_date.strftime('%Y-%m-%d %H:%M:%S')}")
 
-                if from_date.hour == 23:
```

Comment:
Note that this flow is for fetching `statistics` data, not data that would appear in the sensors.
At first, this starts loading from a month back.

`from_date` represents that last datapoint we have fetched data for  -- since the coordinator runs hourly, we can get parts of the day. So - if the last hour is `23` - the next "pull" is for the next day.

Then - if  the last day of fetching is today - we'll just fetch 01:00 hours from today.

Compared to your examples in the thread - this is not in resolution=3 (but hourly(=1)) for Statistics, and also - incorrect in terms of logic